Modules/_io/textio.c

```diff
@@ -2844,10 +2844,15 @@ _io_TextIOWrapper_tell_impl(textio *self)
     /* Fast search for an acceptable start point, close to our
        current pos */
     skip_bytes = (Py_ssize_t) (self->b2cratio * chars_to_skip);
-    skip_back = 1;
-    assert(skip_back <= PyBytes_GET_SIZE(next_input));
-    input = PyBytes_AS_STRING(next_input);
-    while (skip_bytes > 0) {
+    /* Skip the optimization if next_input is empty */
+    if (PyBytes_GET_SIZE(next_input) == 0) {
+        skip_bytes = 0;
+    }
+    else {
+        skip_back = 1;
+        assert(skip_back <= PyBytes_GET_SIZE(next_input));
+        input = PyBytes_AS_STRING(next_input);
+        while (skip_bytes > 0) {
```

Comment:
```suggestion
    }
    while (skip_bytes > 0) {
```

I think it is better - you don't need to ident while body in this this case.